packages/synapse-react-client/src/components/GenericCard/Linkify.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import { TargetEnum } from '../CardContainerLogic'
+import { Link } from '@mui/material'
+import { PRODUCTION_ENDPOINT_CONFIG } from '../../utils/functions/getEndpoint'
+
+export type LinkifyProps = {
+  text?: string
+  className?: string
+}
+
+export type LinkifyRule = {
+  regex: RegExp
+  onMatch: (value: string) => string
+}
+
+const synapseIdRule: LinkifyRule = {
+  regex: /(syn\d+(?:\.\d+)?)/,
+  onMatch: value => `${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${value}`,
+}
+
+const httpRule: LinkifyRule = {
+  regex: /((ftp|http|https):\/\/[^ ",]+)/,
+  onMatch: value => value,
+}
+
+const pubMedRule: LinkifyRule = {
+  regex: /(PMID:\d+)/,
+  onMatch: value => {
+    const id = value.slice('PMID:'.length)
+    return `https://pubmed.ncbi.nlm.nih.gov/${id}`
+  },
+}
+const mutationIdRule: LinkifyRule = {
+  regex: /(MGI:\d+)/,
+  onMatch: value => `https://www.informatics.jax.org/allele/${value}`,
+}
+const sciCrunchResolverRule: LinkifyRule = {
+  regex:
+    /(AB_\d+)|(Addgene_\d+)|(CVCL_[a-zA-Z0-9]+)|(MMRRC_\d+-.{3})|(IMSR_JAX:\d+)/,
+  onMatch: value => `https://scicrunch.org/resolver/${value}`,
+}
+const clinVarVCVRule: LinkifyRule = {
+  regex: /(ClinVar:VCV\d+)/,
+  onMatch: value => {
+    const id = value.slice('ClinVar:'.length)
+    return `https://www.ncbi.nlm.nih.gov/clinvar/variation/${id}`
+  },
+}
+const clinVarRCVRule: LinkifyRule = {
+  regex: /(ClinVar:RCV\d+)/,
+  onMatch: value => {
+    const id = value.slice('ClinVar:'.length)
+    return `https://www.ncbi.nlm.nih.gov/clinvar/${id}`
+  },
+}
+
+const rules: LinkifyRule[] = [
+  httpRule,
+  synapseIdRule,
+  pubMedRule,
+  sciCrunchResolverRule,
+  mutationIdRule,
+  clinVarVCVRule,
+  clinVarRCVRule,
+]
+const splitter = new RegExp(rules.map(r => r.regex.source).join('|'), 'g')
+
+const Linkify: React.FC<LinkifyProps> = ({ text, className }) => {
+  if (text === undefined) {
+    return <></>
+  } else if (!text.split) {
+    return <p>{text}</p>
+  }
+
+  const parts = text.split(splitter)
```

Comment:
Took me a moment and had to look back at the tests to figure out why we needed to split here, so I think a quick note would be helpful
```suggestion
  // Split the string using the joined regular expressions so we can add links to identifiers within free text
  const parts = text.split(splitter)
```

Reply:
Added comment